### PR TITLE
Introduce Quarkus mDNS

### PR DESCRIPTION
**Pull request by melloware**
- Closes https://github.com/quarkusio/quarkus/issues/43214

### CODE REVIEW

**Comment by melloware** [comment]
@gastaldi feel free to fix anything i did wrong 😄

**Comment by gastaldi** [comment]
Isn't the library named `jmdns`? Given that the extension is meant to support that library I think it will be better if it's called like it?

File: terraform-scripts/quarkus-mdns.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Create repository
+resource "github_repository" "quarkus_mdns" {
+  name                   = "quarkus-mdns"
+  description            = "Quarkus mDNS extension - Multicast Domain Name System"
+  homepage_url           = "https://docs.quarkiverse.io/quarkus-mdns/2.0.0"
```
Comment (gastaldi) [inline terraform-scripts/quarkus-mdns.tf:5]:
```suggestion
  homepage_url           = "https://docs.quarkiverse.io/quarkus-jmdns/dev/"
```

**Comment by melloware** [comment]
Well multicast DNS is a generic protocol the library is jmdns but this would give us flexibility to swap it out for a different library in the future.  So I like to generically call it just `mdns`.

File: terraform-scripts/quarkus-mdns.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Create repository
+resource "github_repository" "quarkus_mdns" {
+  name                   = "quarkus-mdns"
+  description            = "Quarkus mDNS extension - Multicast Domain Name System"
+  homepage_url           = "https://docs.quarkiverse.io/quarkus-jmdns/dev/"
```
Comment (gastaldi) [inline terraform-scripts/quarkus-mdns.tf:5]:
```suggestion
  homepage_url           = "https://docs.quarkiverse.io/quarkus-mdns/dev/"
```

**Comment by melloware** [comment]
Its actually had many names in the past `Bonjour/Avahi/Rendezvous/ZeroConf` services

**Comment by gastaldi** [comment]
> Well multicast DNS is a generic protocol the library is jmdns but this would give us flexibility to swap it out for a different library in the future. So I like to generically call it just `mdns`. Apple calls is `Bonjour` before the spec was released and renamed `mDNS`

Right, I thought the extension was meant to provide native support for the `jmdms` library, but I guess that makes sense if the plan is to provide support for mDNS regardless if using this library or not

**Review by gastaldi — APPROVED** [review]
Splendid work! Good job!

**Pull request merged**